solver/llbsolver/mounts/mount.go

```diff
@@ -123,7 +124,18 @@ func (g *cacheRefGetter) getRefCacheDirNoCache(ctx context.Context, key string,
 		}
 		locked := false
 		for _, si := range sis {
-			if mRef, err := g.cm.GetMutable(ctx, si.ID()); err == nil {
+			mRef, err := g.cm.GetMutable(ctx, si.ID())
+			var needsRemoteProviders cache.NeedsRemoteProviderError
+			if errors.As(err, &needsRemoteProviders) && ref != nil {
```

Comment:
As calling `descHandlers` does not seem expensive I would add it always, instead of checking for error.